elevate-cpanel

```diff
@@ -3296,10 +3293,29 @@ sub read_redhat_release() {
 
 sub ssystem (@args) {
     INFO( "Running: " . join( " ", @args ) );
-    print "\n";    # Buffer so they can more easily read the output.
-    my $ret = system(@args);
-    print "\n";    # Buffer so they can more easily read the output.
-    return $ret;
+    INFO();    # Buffer so they can more easily read the output.
+
+    # emulate shell behavior of system()
+    if ( scalar @args == 1 && $args[0] =~ m/[\$&*(){}\[\]'";\\|?<>~`\n]/ ) {
+        unshift @args, qw(/usr/bin/bash -c);
+    }
+
+    my $program = shift @args;
+    my ( $callback_out, $callback_err ) = map {
+        my $label = $_;
+        Cpanel::IOCallbackWriteLine->new( sub ($line) { chomp $line; INFO("{$label} $line"); } )
+    } qw(stdout stderr);
+    my $sr = Cpanel::SafeRun::Object->new(
+        program      => $program,
+        args         => [@args],
+        stdout       => $callback_out,
+        stderr       => $callback_err,
+        timeout      => 0,
+        read_timeout => 0,
+    );
+    INFO();    # Buffer so they can more easily read the output.
+
+    return $? = $sr->CHILD_ERROR;    ## no critic qw(Variables::RequireLocalizedPunctuationVars) -- emulate return behavior of system()
```

Comment:
I'm not wild about us clobbering $? why would we want to set that?

Reply:
The use of `system` already clobbered `$?`, so this just maintains "feature" parity. It doesn't need to stay.